library/core/src/slice/sort/unstable/quicksort.rs

```diff
@@ -98,13 +101,15 @@ where
         return 0;
     }
 
-    // Allows for panic-free code-gen by proving this property to the compiler.
     if pivot >= len {
         intrinsics::abort();
     }
 
-    // Place the pivot at the beginning of slice.
-    v.swap(0, pivot);
+    // SAFETY: We checked that `pivot` is in-bounds.
+    unsafe {
+        // Place the pivot at the beginning of slice.
+        v.swap_unchecked(0, pivot);
```

Comment:
Is the abort above no longer sufficient to prove in-bounds for some reason?

Reply:
I'm afraid it never was, I looked at this code again with godbolt and the compiler didn't omit the panics before. IIRC I inspected that piece of code in a full binary disassembly in the past, not sure what my process error was, but doing `swap_unchecked` allows us to be sure, and guard against future compilers optimizing differently.